```
go: downloading github.com/aristanetworks/goarista v0.0.0-20170210015632-ea17b1a17847
go: downloading golang.org/x/sys v0.0.0-[20](https://github.com/nervosnetwork/ckb-sdk-go/actions/runs/10314934806/job/28555511295?pr=221#step:4:21)220422013727-9388b58f7150
ok  	github.com/nervosnetwork/ckb-sdk-go/v2/address	0.026s
# github.com/nervosnetwork/ckb-sdk-go/v2/transaction/signer_test [github.com/nervosnetwork/ckb-sdk-go/v2/transaction/signer.test]
Error: transaction/signer/example_test.go:20:44: not enough arguments in call to ctx.rpc.GetLiveCell
	have (context.Context, *types.OutPoint, bool)
	want (context.Context, *types.OutPoint, bool, *bool)
ok  	github.com/nervosnetwork/ckb-sdk-go/v2/collector	2.405s
```

Don't know why the test failed. I can't find `example_test.go` file.

--- a/collector/builder/dao.go
+++ b/collector/builder/dao.go
@@ -47,7 +47,7 @@ func NewDaoTransactionBuilder(network types.Network, iterator collector.CellIter
 	depositCellCapacity := uint64(0)
 	reward := uint64(0)
 	if transactionType == DaoTransactionTypeWithdraw {
-		txWithStatus, err := client.GetTransaction(context.Background(), daoOutPoint.TxHash)
+		txWithStatus, err := client.GetTransaction(context.Background(), daoOutPoint.TxHash, nil)
 		if err != nil {
 			return nil, err
 		}
@@ -94,7 +94,7 @@ func getTransactionType(outputData []byte) (DaoTransactionType, error) {
 }
 
 func getDaoReward(withdrawOutPoint *types.OutPoint, client rpc.Client) (uint64, error) {
-	txWithStatus, err := client.GetTransaction(context.Background(), withdrawOutPoint.TxHash)
+	txWithStatus, err := client.GetTransaction(context.Background(), withdrawOutPoint.TxHash, nil)
 	if err != nil {
 		return 0, err
 	}
@@ -107,7 +107,7 @@ func getDaoReward(withdrawOutPoint *types.OutPoint, client rpc.Client) (uint64, 
 	)
 	for i := 0; i < len(withdrawTx.Inputs); i++ {
 		outPoint := withdrawTx.Inputs[i].PreviousOutput
-		txWithStatus, err := client.GetTransaction(context.Background(), outPoint.TxHash)
+		txWithStatus, err := client.GetTransaction(context.Background(), outPoint.TxHash, nil)
 		if err != nil {
 			return 0, err
 		}

--- a/collector/handler/dao.go
+++ b/collector/handler/dao.go
@@ -102,7 +102,7 @@ type ClaimInfo struct {
 }
 
 func NewClaimInfo(client rpc.Client, withdrawOutpoint *types.OutPoint) (*ClaimInfo, error) {
-	txWithStatus, err := client.GetTransaction(context.Background(), withdrawOutpoint.TxHash)
+	txWithStatus, err := client.GetTransaction(context.Background(), withdrawOutpoint.TxHash, nil)
 	if err != nil {
 		return nil, err
 	}
@@ -111,7 +111,7 @@ func NewClaimInfo(client rpc.Client, withdrawOutpoint *types.OutPoint) (*ClaimIn
 	var depositBlockHash types.Hash
 	for i := 0; i < len(withdrawTx.Inputs); i++ {
 		outPoint := withdrawTx.Inputs[i].PreviousOutput
-		txWithStatus, err := client.GetTransaction(context.Background(), outPoint.TxHash)
+		txWithStatus, err := client.GetTransaction(context.Background(), outPoint.TxHash, nil)
 		if err != nil {
 			return nil, err
 		}
@@ -175,7 +175,7 @@ type WithdrawInfo struct {
 }
 
 func NewWithdrawInfo(client rpc.Client, depositOutPoint *types.OutPoint) (*WithdrawInfo, error) {
-	txWithStatus, err := client.GetTransaction(context.Background(), depositOutPoint.TxHash)
+	txWithStatus, err := client.GetTransaction(context.Background(), depositOutPoint.TxHash, nil)
 	if err != nil {
 		return nil, err
 	}

--- a/dao/helper.go
+++ b/dao/helper.go
@@ -33,7 +33,7 @@ func (c *DaoHelper) GetDaoDepositCellInfo(outpoint *types.OutPoint, withdrawBloc
 
 // GetDaoDepositCellInfoWithWithdrawOutpoint Get information for DAO cell deposited as outpoint and withdrawn in block where the withdrawCellOutPoint is
 func (c *DaoHelper) GetDaoDepositCellInfoWithWithdrawOutpoint(outpoint *types.OutPoint, withdrawCellOutPoint *types.OutPoint) (DaoDepositCellInfo, error) {
-	withdrawTransaction, err := c.Client.GetTransaction(context.Background(), withdrawCellOutPoint.TxHash)
+	withdrawTransaction, err := c.Client.GetTransaction(context.Background(), withdrawCellOutPoint.TxHash, nil)
 	if err != nil {
 		return DaoDepositCellInfo{}, err
 	}
@@ -51,7 +51,7 @@ func (c *DaoHelper) GetDaoDepositCellInfoByNow(outpoint *types.OutPoint) (DaoDep
 
 // getDaoDepositCellInfo Get information for DAO cell deposited as outpoint and withdrawn in withdrawBlock
 func (c *DaoHelper) getDaoDepositCellInfo(outpoint *types.OutPoint, withdrawBlockHeader *types.Header) (DaoDepositCellInfo, error) {
-	depositTransactionWithStatus, err := c.Client.GetTransaction(context.Background(), outpoint.TxHash)
+	depositTransactionWithStatus, err := c.Client.GetTransaction(context.Background(), outpoint.TxHash, nil)
 	if err != nil {
 		return DaoDepositCellInfo{}, err
 	}

--- a/rpc/client.go
+++ b/rpc/client.go
@@ -98,8 +98,11 @@ type Client interface {
 	// Note that the given block is included in the median time. The included block number range is [MAX(block - 36, 0), block].
 	GetBlockMedianTime(ctx context.Context, blockHash types.Hash) (uint64, error)
 
-	// GetFeeRateStatics Returns the fee_rate statistics of confirmed blocks on the chain
+	// Deprecated: use GetFeeRateStatistics instead
 	GetFeeRateStatics(ctx context.Context, target interface{}) (*types.FeeRateStatics, error)
+
+	// GetFeeRateStatistics Returns the fee_rate statistics of confirmed blocks on the chain
+	GetFeeRateStatistics(ctx context.Context, target interface{}) (*types.FeeRateStatistics, error)
 
 	////// Experiment
 	// DryRunTransaction dry run transaction and return the execution cycles.
@@ -535,15 +538,19 @@ func (cli *client) GetBlockMedianTime(ctx context.Context, blockHash types.Hash)
 }
 
 func (cli *client) GetFeeRateStatics(ctx context.Context, target interface{}) (*types.FeeRateStatics, error) {
-	var result types.FeeRateStatics
+	return cli.GetFeeRateStatistics(ctx, target)
+}
+
+func (cli *client) GetFeeRateStatistics(ctx context.Context, target interface{}) (*types.FeeRateStatistics, error) {
+	var result types.FeeRateStatistics
 	switch target := target.(type) {
 	case nil:
-		if err := cli.c.CallContext(ctx, &result, "get_fee_rate_statics", nil); err != nil {
+		if err := cli.c.CallContext(ctx, &result, "get_fee_rate_statistics", nil); err != nil {
 			return nil, err
 		}
 		break
 	case uint64:
-		if err := cli.c.CallContext(ctx, &result, "get_fee_rate_statics", hexutil.Uint64(target)); err != nil {
+		if err := cli.c.CallContext(ctx, &result, "get_fee_rate_statistics", hexutil.Uint64(target)); err != nil {
 			return nil, err
 		}
 		break

--- a/rpc/client.go
+++ b/rpc/client.go
@@ -64,7 +64,7 @@ type Client interface {
 	GetLiveCell(ctx context.Context, outPoint *types.OutPoint, withData bool) (*types.CellWithStatus, error)
 
 	// GetTransaction returns the information about a transaction requested by transaction hash.
-	GetTransaction(ctx context.Context, hash types.Hash) (*types.TransactionWithStatus, error)
+	GetTransaction(ctx context.Context, hash types.Hash, only_committed *bool) (*types.TransactionWithStatus, error)
 
 	// GetBlockEconomicState return block economic state, It includes the rewards details and when it is finalized.
 	GetBlockEconomicState(ctx context.Context, hash types.Hash) (*types.BlockEconomicState, error)
@@ -446,9 +446,14 @@ func (cli *client) GetLiveCell(ctx context.Context, point *types.OutPoint, withD
 	return &result, err
 }
 
-func (cli *client) GetTransaction(ctx context.Context, hash types.Hash) (*types.TransactionWithStatus, error) {
+func (cli *client) GetTransaction(ctx context.Context, hash types.Hash, only_committed *bool) (*types.TransactionWithStatus, error) {
 	var result types.TransactionWithStatus
-	err := cli.c.CallContext(ctx, &result, "get_transaction", hash)
+	var err error
+	if only_committed == nil {
+		err = cli.c.CallContext(ctx, &result, "get_transaction", hash)
+	} else {
+		err = cli.c.CallContext(ctx, &result, "get_transaction", hash, *only_committed)
+	}
 	if err != nil {
 		return nil, err
 	}

--- a/rpc/client_test.go
+++ b/rpc/client_test.go
@@ -488,10 +488,10 @@ func TestGetTransactionsGrouped(t *testing.T) {
 }
 
 func TestClient_GetFeeRateStatics(t *testing.T) {
-	statics, err := testClient.GetFeeRateStatics(context.Background(), nil)
+	statics, err := testClient.GetFeeRateStatistics(context.Background(), nil)
 	assert.NoError(t, err)
 	assert.NotNil(t, statics)
-	statics2, err := testClient.GetFeeRateStatics(context.Background(), 1)
+	statics2, err := testClient.GetFeeRateStatistics(context.Background(), 1)
 	assert.NoError(t, err)
 	assert.NotNil(t, statics2)
 }

--- a/rpc/client_test.go
+++ b/rpc/client_test.go
@@ -99,7 +99,7 @@ func TestClient_GetBlockWithCycles(t *testing.T) {
 
 func TestClient_GetTransaction(t *testing.T) {
 	txView, err := testClient.GetTransaction(ctx,
-		types.HexToHash("0x8277d74d33850581f8d843613ded0c2a1722dec0e87e748f45c115dfb14210f1"))
+		types.HexToHash("0x8277d74d33850581f8d843613ded0c2a1722dec0e87e748f45c115dfb14210f1"), nil)
 	assert.NoError(t, err)
 	tx := txView.Transaction
 	status := txView.TxStatus

--- a/types/chain.go
+++ b/types/chain.go
@@ -326,10 +326,13 @@ type EstimateCycles struct {
 	Cycles uint64 `json:"cycles"`
 }
 
-type FeeRateStatics struct {
+type FeeRateStatistics struct {
 	Mean   uint64 `json:"mean"`
 	Median uint64 `json:"median"`
 }
+
+// Deprecated: Use FeeRateStatistics instead
+type FeeRateStatics = FeeRateStatistics
 
 type TransactionAndWitnessProof struct {
 	BlockHash         Hash   `json:"block_hash"`

--- a/types/stats.go
+++ b/types/stats.go
@@ -51,7 +51,7 @@ type DeploymentInfo struct {
 	Timeout            uint64
 	MinActivationEpoch uint64
 	Period             uint64
-	Threshold          RationalU256
+	Threshold          jsonRationalU256
 	Since              uint64
 	State              DeploymentState
 }

